server/validation/campaign.js

```diff
@@ -0,0 +1,32 @@
+const User = require("../models/user");
+const Campaign = require("../models/campaign");
+
+exports.validateCampaignInput = async data => {
+  const errors = {};
+  const { name, ownedBy } = data;
+
+  if (!name) errors.name = "name is required";
+
+  const userExists = await User.findById(ownedBy);
+  if (!userExists) errors.ownedBy = "not owned by a valid user";
+
+  return {
+    errors,
+    isValid: !Object.keys(errors).length > 0
+  };
+};
+
+exports.validateAddProspectsInput = (campaignId, prospectIds) => {
+  // check for prospoects already in the campaign
+  const uniqueProspects = [];
+
+  prospectIds.forEach(async prospectId => {
+    const prospectExists = await Campaign.findOne({
+      _id: campaignId,
+      "prospects.prospectId": prospectId
+    });
+    if (!prospectExists) uniqueProspects.push(prospectId);
```

Comment:
This is a pretty inefficient validator, I'd rather you select the campaign once (since there is only one campaign Id, and then see if the prospects are in that campaign in javascript 

However, I noticed that's a similar logic to what you are doing already in the API call, so this validator is not necessary

Reply:
I moved this into the API call and selected the campaign only once. 

I also noticed I wasn't checking if the prospect is owned by the user before adding, so have also added that check. Rather than fetching all prospects owned by the user to check, I'm making one call to the db per prospect being added. My thinking is this will be faster as prospect lists get large, but maybe I'm wrong?

Reply:
You could possible fetch all the prospects at once though given a list of ids - that would probably be the most optimal